### PR TITLE
fix #281253: staff spacer down ignored on bottom of page

### PR DESCRIPTION
**Pull request by MarcSabatella**
Resolves: https://musescore.org/en/node/281253

A staff spacer on the bottom system of a page does not work correctly:
we are looking foir a spacer on the *first* staff rather than the *last*.
So a spacer on the last staff is ignored, and one on the first is used
when it shouldn't be.

The code to calculate the amount of spacer required below the last
system of a page relies on System::lastVisibleSysStaff(),
which is return the correct SysStaff, but we are then attempting
to access the idx member, which has never been set up and is always 0.
Fix is to initialize idx when setting up the list of staves.

vtest created

Note: I originally fixed this issue several months ago, but the fix only worked for single-staff systems, because idx was always 0.

### CODE REVIEW

File: libmscore/system.cpp [hunk 1]
```diff
@@ -162,6 +162,7 @@ Box* System::vbox() const
 SysStaff* System::insertStaff(int idx)
       {
       SysStaff* staff = new SysStaff;
+      staff->idx = idx;
```
Comment (dmitrio95) [inline libmscore/system.cpp]: In order to maintain `SysStaff::idx` value correctly this method should probably check whether it appends a staff to the end of staves list or inserts it somewhere in the middle and shift other indices staves correspondingly in the latter case. Something similar should probably be done in `System::removeStaff()`.

Still if the only place where this `idx` is used is here:
https://github.com/musescore/MuseScore/blob/3b9cf4d961020a62bec3aa0671bcef672edad341/libmscore/system.cpp#L1352
then maybe we could get rid of `idx` entirely and make `firstVisibleSysStaff`/`lastVisibleSysStaff` (or some similar functions) tell us the staff index directly based on its position in staves list. Then we wouldn't need to maintain `idx` value ourselves.

Comment (MarcSabatella) [reply libmscore/system.cpp]: Right now this list is never modified, only rebuilt from scratch, but yeah, better safe than sorry.  It's also possible, btw, to set idx during layout when it's even safer.   Somehow it does seems like there should be a way to get from a SyStaff back to the actual staff, even if right now this is the only place we seem to need it.  But, maybe not something I need to solve, changing the other functions to return the index is probably the better way to go for now.  Will do.

**Comment by MarcSabatella** [comment]
Updated as discussed.

**Comment by dmitrio95** [comment]
Did you push the changes? It looks like they are not in this PR currently.

**Comment by MarcSabatella** [comment]
Ooops, sorry, pushed now.

**Pull request merged**